php/classes/DB.php

```diff
@@ -0,0 +1,108 @@
+        if (isset($_POST['checkReceiptAndSelectNext'])) {
+            $isChecked = 1;
+        }
+
+        if ($_POST['receiptid'] == "") { // new receipt
+            $sth = $this->pdo->prepare('INSERT INTO `receipts`(`photo_name`, `photo_data`, `date_emission`, `category`, `provider`, `montant_ttc`, `tva`, `checked`, `description`) VALUES (:photo_name, :photo_data, :date, :receiptCategory, :provider, :amountTTC, :tva, :isChecked, :description)');
+            
+            $photoData = file_get_contents($_FILES['photo']['tmp_name']);
+            $sth->bindParam(':photo_name',  $_FILES["photo"]["name"], PDO::PARAM_STR);
+            $sth->bindParam(':photo_data', $photoData);
+
+        }else {  // update receipt
+            if ( ($_FILES['photo']['tmp_name'] != '') && ($_FILES['photo']['tmp_name'] != '') ){ // if new photo 
+                $sth = $this->pdo->prepare('UPDATE `receipts` SET `photo_name`=:photo_name, `photo_data`=:photo_data, `date_emission`=:date, `category`=:receiptCategory, `provider`=:provider, `montant_ttc`=:amountTTC, `tva`=:tva, `checked`=:isChecked, `description`=:description WHERE `receipts`.`id` = :id');
+                $photoData = file_get_contents($_FILES['photo']['tmp_name']);
+                $sth->bindParam(':photo_name',  $_FILES["photo"]["name"], PDO::PARAM_STR);
+                $sth->bindParam(':photo_data', $photoData);   
+            } else { // no new photo
+                $sth = $this->pdo->prepare('UPDATE `receipts` SET `date_emission`=:date, `category`=:receiptCategory, `provider`=:provider, `montant_ttc`=:amountTTC, `tva`=:tva, `checked`=:isChecked, `description`=:description WHERE `receipts`.`id` = :id');
+            }
+            $sth->bindParam(':id', $_POST['receiptid'], PDO::PARAM_STR);  
+        }
+
+
+        $sth->bindParam(':date', $_POST['date'], PDO::PARAM_STR);
+        $sth->bindParam(':receiptCategory', $_POST['receiptCategory'], PDO::PARAM_STR);
+        $sth->bindParam(':provider', $_POST['provider'], PDO::PARAM_STR);
+        $sth->bindParam(':amountTTC', $_POST['amountTTC'], PDO::PARAM_INT);
+        $sth->bindParam(':tva', $_POST['tva'], PDO::PARAM_STR);
+        $sth->bindParam(':isChecked', $isChecked, PDO::PARAM_STR);
+        $sth->bindParam(':description', $_POST['description'], PDO::PARAM_STR);
+        $result = $sth->execute();
+
+
+    } catch (Exception $ex) {
+        $this->msg = sendMessage($ex->getMessage(), 'danger');
+        return false;
+    }
+    $this->msg= sendMessage("Ticket enregistré ! ", "success");
+    return true;
+  }
+
+ 
+  //  get photo name
+  function getImg ($name) {
+    $this->stmt = $this->pdo->prepare(
+      "SELECT `photo_data` FROM `receipts` WHERE `photo_name`=?"
+    );
+    $this->stmt->execute([$name]);
+    $img = $this->stmt->fetch();
+    return $img['photo_data'];
+  }
+  
+  //  generate src for update receipt photo preload
+  function getImgSrc ($name) {
+    $img = base64_encode($this->getImg($name));
```

Comment:
This has nothing to do with the database, so either you class is not about handling the database, or this function does not belong here